main.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+)
+
+func main() {
+	buf := bufio.NewReader(os.Stdin)
+	var checkEquality []byte
+
+	for {
+		content, readErr := buf.ReadBytes('\n')
+		if readErr != nil && readErr != io.EOF {
+			fmt.Fprintln(os.Stderr, readErr)
+			os.Exit(1)
+		}
+
+		if len(content) != 0 && content[len(content)-1] != 0xa {
```

Comment:
Isn't there any built-in function to check this?

Reply:
there probably is but from "scratch" is always better

Reply:
It's always... worse.

Reply:
why am I re-developing the linux command uniq then :p

Reply:
To learn something.

Reply:
Ok but which built-in function do mean?

Reply:
What are you checking for?

Reply:
the first conditions checks if there already is a newline and the second checks if the last element in the array is a newline or not

Reply:
In short, you're checking whether content is **suffix**ed with `\n`. Any idea which function I mean?